### ISSUE TITLE
reveal function

### ISSUE THREAD

**Issue by giorgiobeggiora** [lixinliang/tree-view-search-bar#18]
Can you please add the following "filter" option? thanx :)

after search (i suggest live typing = show results for each keystroke) results are accessible with arrow keys. Firs attempt to open (enter) a file or a folder reveal it instead, and clears the search query.

Thanks!

**Comment by lixinliang** [comment]
Thank you for your suggestion.

I am wonder that your idea is, after search the plugin will locate to the file and folder ( and open like using enter ), and then, clear the search query..

Is the description correct?

**Comment by giorgiobeggiora** [comment]
yes, however i though more about my idea and improved it a little bit, what do you think about it?
I call "query tag" the green rectangle containing characters and the "x" to close.

- the search is executed at every keystroke, because a "query tag" is automatically created and updated.
- pressing "Enter" is not needed to search: it only starts a new "query tag".
- "ESC" clears the last "query tag"
- the plugin will show the results (files and folders)
- when i single click on one result, a "soft tab" is opened
- when i double click on one result, a "hard tab" is opened, the search query is cleared and the tree view reveal the folder or file
- when i double click on one result while holding CTRL or CMD, a "hard tab" is opened

Having all these actions as options in a panel should be great

**Comment by giorgiobeggiora** [comment]
Regarding the "arrow keys" functionality i was talking about, it's almost implemented in Atom. 
My idea is:

- (DONE) a shortcut to focus the tree view is needed (default Atom feature: CTRL-0 = toggles focus between tree-view and editor)
- (TO DO) a shortcut to focus search bar is needed (maybe CTRL+SHIFT+0 = toggles focus between search field and tree view ?).
- (DONE) the tree must be navigable via arrow keys, and enter should open files (default tree-view feature)

**Comment by lixinliang** [comment]
Yeah~, there are a lots of good ideas about the "tag input".

At that time, when I was coding for this feature, I used a plugin to help me for quickly development.
https://github.com/lixinliang/tree-view-search-bar/blob/dev/package.json#L33

Be honest, if any cool and powerful tag-input plugin, I think it's easy to implement.
Otherwise, I have to make it self 😂   such as soft tab / hard tab ... emmm...

And to key actions, I am going to use mousetrap.
https://craig.is/killing/mice

About the atom shortcut, I will complete soon.
https://github.com/lixinliang/tree-view-search-bar/blob/0.1.x/keymaps/main.cson
I did it before.

And the "arrow keys" functionality, maybe I should read the atom doc to figure out first.

I will do the list as soon as 🙈   thanks
